Source/WebKit/WebProcess/cocoa/VideoFullscreenManager.mm

```diff
@@ -335,24 +346,27 @@ static FloatRect inlineVideoFrame(HTMLVideoElement& element)
     interface->setAnimationState(VideoFullscreenInterfaceContext::AnimationType::IntoFullscreen);
 
     bool allowsPictureInPicture = videoElement.webkitSupportsPresentationMode(HTMLVideoElement::VideoPresentationMode::PictureInPicture);
-    
-    if (videoElement.document().settings().blockMediaLayerRehostingInWebContentProcess())
-        m_page->send(Messages::VideoFullscreenManagerProxy::SetupFullscreenWithID(contextId, videoElement.layerHostingContextID(), videoRect, FloatSize(videoElement.videoWidth(), videoElement.videoHeight()), m_page->deviceScaleFactor(), interface->fullscreenMode(), allowsPictureInPicture, standby, videoElement.document().quirks().blocksReturnToFullscreenFromPictureInPictureQuirk()));
-    else {
-        if (!interface->layerHostingContext()->rootLayer()) {
-            auto videoLayer = model->createVideoFullscreenLayer();
-            [videoLayer setDelegate:[WebActionDisablingCALayerDelegate shared]];
-            [videoLayer setName:@"Web Video Fullscreen Layer"];
-            [videoLayer setAnchorPoint:CGPointMake(0, 0)];
-            [videoLayer setPosition:CGPointMake(0, 0)];
-            [videoLayer setBackgroundColor:cachedCGColor(WebCore::Color::transparentBlack).get()];
 
+    if (!interface->rootLayer()) {
```

Comment:
Why is it ok to create a root layer if we're blocking media layer reposting?

Reply:
> Why is it ok to create a root layer if we're blocking media layer reposting?

I believe it is ok to create a root layer in the WebContent process when blocking blocking media layer re-hosting, since this will not require access to CARenderServer in the sandbox. Ideally, we would also remove the layer tree, but since there is caption code that currently depends on this tree to be present to get the caption bounds correct, I kept it. I think we could address this in a follow-up patch, would you be ok with that?

Thanks for reviewing!